DependencyInjection/Compiler/MappingPass.php

```diff
@@ -11,85 +11,101 @@
-            );
-
-            $container->setDefinition(sprintf('es.manager.%s', $managerName), $managerDefinition);
-
-            // Make es.manager.default as es.manager service.
-            if ($managerName === 'default') {
-                $container->setAlias('es.manager', 'es.manager.default');
-
-                if (Kernel::MAJOR_VERSION >= 4) {
-                    $container->getAlias('es.manager')
-                        ->setPublic(true)
-                    ;
+        $parser = $container->get(DocumentParser::class);
+
+        $indexes = [];
+        $defaultIndex = null;
+        foreach ($this->getNamespaces(
+            $container->getParameter('kernel.project_dir')
+                .$container->getParameter(Configuration::ONGR_SOURCE_DIR)
+        ) as $namespace) {
+            $indexMapping = $parser->getIndexMetadata($namespace);
+            $indexAlias = $parser->getIndexAliasName($namespace);
+            if (!empty($indexMapping)) {
+                $indexServiceDefinition = new Definition(IndexService::class, [
+                    $namespace,
+                    $container->getDefinition(Converter::class),
+                    $container->getDefinition(DocumentParser::class),
+                    $container->getDefinition('event_dispatcher'),
+                    $container->getDefinition('serializer'),
+                    $container->getParameter(Configuration::ONGR_PROFILER_CONFIG)
+                        ? $container->getDefinition('ongr.esb.tracer') : null
+                ]);
+                $indexServiceDefinition->setPublic(true);
+
+                $container->setDefinition($namespace, $indexServiceDefinition);
+                $indexes[$indexAlias] = $namespace;
+                $isCurrentIndexDefault = $parser->isDefaultIndex($namespace);
+                if ($defaultIndex && $isCurrentIndexDefault) {
+                    throw new \RuntimeException(
+                        sprintf(
+                            'Only one index can be set as default. We found 2 indexes as default ones `%s` and `%s`',
+                            $defaultIndex,
+                            $indexAlias
+                        )
+                    );
+                }
+
+                if ($isCurrentIndexDefault) {
+                    $defaultIndex = $indexAlias;
                 }
             }
+        }
 
-            $mappings = $collector->getMappings($manager['mappings']);
+        $container->setParameter(Configuration::ONGR_INDEXES, $indexes);
 
-            // Building repository services.
-            foreach ($mappings as $repositoryType => $repositoryDetails) {
-                $repositoryDefinition = new Definition(
-                    'ONGR\ElasticsearchBundle\Service\Repository',
-                    [$repositoryDetails['namespace']]
-                );
-                $repositoryDefinition->setPublic(true);
+        $defaultIndex = $defaultIndex ?? array_shift(array_keys($indexes));
```

Comment:
If no index is present (f.e. in a new environment) `array_shift(array_keys($indexes))` is invalid.

Reply:
That is completely true. Thanks for the notice.